src/main/java/vg/civcraft/mc/civmodcore/ACivMod.java

```diff
@@ -23,7 +24,7 @@
 import vg.civcraft.mc.civmodcore.serialization.NBTSerialization;
 import vg.civcraft.mc.civmodcore.util.Iteration;
 
-public abstract class ACivMod extends JavaPlugin {
+public abstract class ACivMod extends JavaPlugin implements Closeable {
```

Comment:
It effectively provides an `onUnload()` method, though it's not as good as Java 9's [Cleanable](https://docs.oracle.com/javase/9/docs/api/java/lang/ref/Cleaner.Cleanable.html), but we haven't officially moved from Java 8 yet. This is a pretty minor change that shouldn't effect people via changed signatures.

Reply:
But there already is an onUnload? What does this add or how does this improve anything

Reply:
Because `onDisable()` and `onUnload()` are two separate things. Just because the plugin has been disabled doesn't mean the instance no longer exists or has been abandoned. It allows a greater level of control without adding any complexity since no plugin is forced to use this.

Reply:
Why would you ever disable a plugin except during startup and shutdown? That's bound to cause countless other problems and kill the server. We explicitly don't support any kind of runtime reload shenanigans (with some plugins providing their own exception).

Reply:
Yes, and how is not supporting runtime reloading doing for you in terms of testing?